crypto/x509/v3_ncons.c

```diff
@@ -581,13 +595,59 @@ static int nc_dns(ASN1_IA5STRING *dns, ASN1_IA5STRING *base)
 
 }
 
+/*
+ * This function implements comparison between ASCII/U-label in eml
+ * and A-label in base according to RFC 8398, section 6.
+ * Convert base to U-label and ASCII-parts of domain names, for base
+ * Octet-to-octet comparison of `eml` and `base` hostname parts
+ * (ASCII-parts should be compared in case-insensitive manner)
```

Comment:
Apologies I'm not following the terminology for "`eml` and `base` hostname" . Perhaps local-part and domain part?

Reply:
`eml` is the hostname from the email specified.
`base` is the hostname from the name constraint.